### ISSUE TITLE
[BUG] NestedSet behaviour: appendTo() behaves as prependTo().

### ISSUE THREAD

**Issue by temuri416** [phalcon/incubator#513]
Hi,

I think there is a bug in inserting nodes to parent. 

All newly inserted nodes are created as parent's first node (or, before any other sibling), regardless of whether **appendTo()** or **prependTo()** is used.

Here's how the rows look like as a result of **appendTo()** or **prependTo()**:

![image](https://cloud.githubusercontent.com/assets/3987203/11856240/3bb7e10a-a41d-11e5-935e-bc584620e810.png)

As you can see node ID#16 was inserted last, however its *lft* value is lower than of its sibling ID#2 which was inserted before.

Thoughts?

Thanks!

**Comment by temuri416** [comment]
@sergeyklay 

Hi there,

Is there a chance that this gets addressed?

Thanks!

**Comment by sergeyklay** [comment]
Hi, I'll try to deal with this in the near future

**Comment by temuri416** [comment]
Awesome, most appreciated!

**Comment by sergeyklay** [comment]
HI, Provide please script to reproduce (including table schema)

**Comment by temuri416** [comment]
Sure, I can put something together.

But aren't you able to reproduce it with your own code and schema? Cuz that's exactly what I was doing. The same applies to #534.
 
Thanks!

**Comment by sergeyklay** [comment]
Unfortunately I still haven't found enough time for this. But I'll try to deal at the weekend. 
In any case, I will need to reproduce the issue.

**Comment by temuri416** [comment]
ok, I'll try to make it easy for you :-) Will post code ASAP.

**Comment by temuri416** [comment]
Hi @sergeyklay ,

Looks like this one is not fixed yet.

Tested with branch `2.0.x`, git commit ID 0bbf7b56c36f7678f4ae4de79b748710035442b4.

Table:

```sql
DROP TABLE IF EXISTS `tree`;
CREATE TABLE IF NOT EXISTS `tree` (
  `id` int(11) NOT NULL,
  `root` int(11) DEFAULT NULL,
  `title` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
  `lft` int(11) DEFAULT NULL,
  `rgt` int(11) DEFAULT NULL,
  `level` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;

ALTER TABLE `tree` ADD PRIMARY KEY (`id`);
ALTER TABLE `tree` MODIFY `id` int(11) NOT NULL AUTO_INCREMENT;
```

Model class:

```php
class Node extends \Phalcon\Mvc\Model
{
    public $title;

    public function initialize()
    {
        $this->setSource('tree');
        $this->addBehavior(new Phalcon\Mvc\Model\Behavior\NestedSet([
            'leftAttribute' => 'lft',
            'rightAttribute' => 'rgt',
            'levelAttribute' => 'level',
            'hasManyRoots' => true,
            'rootAttribute' => 'root',
        ]));
    }
}
```

The code:

```php
$db = new \Phalcon\Db\Adapter\Pdo\Mysql([
    'host' => '127.0.0.1',
    'username' => 'root',
    'password' => 'letmein',
    'dbname' => 'test',
]);

$db->query('truncate table tree');
$db->query('ALTER TABLE tree AUTO_INCREMENT=1');

$this->di['db'] = $db;

$root = new Node;
$root->title = 'ROOT';
$root->saveNode();

// Add "A" as child of root
$node1=new Node;
$node1->title='A';
$node1->appendTo($root);

// Add "B" as last child of root - should be below A
$node2=new Node;
$node2->title='B';
$node2->appendTo($root);

// Add "C" as first child of root - should be above A
$node3=new Node;
$node3->title='C';
$node3->prependTo($root);
```

**Observed result the DB**:

![image](https://cloud.githubusercontent.com/assets/3987203/13384113/8d93edb6-de60-11e5-8f08-c86326bec757.png)

**Observed visual tree structure**:
```
  \ROOT
    |
    +---C
    |
    +---B
    |
    +---A
```

**Expected result the DB**:

![image](https://cloud.githubusercontent.com/assets/3987203/13384138/c97bce20-de60-11e5-9274-fbb01c8255ee.png)

**Expected result the DB**:

```
  \ROOT
    |
    +---C
    |
    +---A
    |
    +---B
```

I'll look at #552 tomorrow.

Thanks!

**Comment by sergeyklay** [comment]
I've realized how it should work. 

```php
$root = new Node;
$root->title = 'ROOT';
$root->saveNode();

// Add "A" as child of root
$node1 = new Node;
$node1->title = 'A';
$node1->appendTo($root);

// Add "B" as last child of root - should be below A
$node2 = new Node;
$node2->title = 'B';
$node2->appendTo($root);

// Add "C" as first child of root - should be above A
$node3 = new Node;
$node3->title = 'C';
$node3->prependTo($root);
```

This code should create such tree:
![nested_set_appendpepend_2](https://cloud.githubusercontent.com/assets/1256298/13413677/038574a0-df55-11e5-87d9-77f5bcf1e2eb.png)

Red numbers - PK
Blue numbers - Level
Black numbers - Left / Right

I fixed this issue in #552
See `NestedSetTest::testShouldAddBelowAndAbove`

**Comment by temuri416** [comment]
@sergeyklay 

I'm a little confused. You're saying it's fixed, yet I don't see any commits since 25th of February, 0bbf7b56c36f7678f4ae4de79b748710035442b4.

https://github.com/phalcon/incubator/pull/552 isn't merged yet, how should I test it?

**Comment by temuri416** [comment]
I've tested with #552 and tree looks as expected.

**Comment by sergeyklay** [comment]
Also check #535 and please #534

**Comment by sergeyklay** [comment]
Fixed in `2.0.x`
